internal/server/presentation/log_test.go

```diff
@@ -129,6 +130,87 @@ func TestNewAMQPLogHandler(t *testing.T) {
 	}
 }
 
+func TestNewAMQPCTRLogHandler(t *testing.T) {
+	fixedTime := time.Date(2024, 9, 23, 23, 7, 32, 840757000, time.UTC)
+
+	ctrLogRequest, msg := testCTRMsg(t, fixedTime)
+
+	tests := []struct {
+		name               string
+		msg                amqp.Delivery
+		mockFunc           func(m *usecase.MockIInsertCTRLogUseCase)
+		expectedStatusCode int
+	}{
+		{
+			name: "success",
+			msg:  msg,
+			mockFunc: func(m *usecase.MockIInsertCTRLogUseCase) {
+				m.EXPECT().InsertCTRLog(
+					gomock.Any(),
+					&usecase.InsertCTRLogDto{
+						EventType: ctrLogRequest.EventType,
+						CreatedAt: ctrLogRequest.CreatedAt,
+						ObjectID:  ctrLogRequest.ObjectID,
+					},
+				).Times(1).Return(nil)
+			},
+			expectedStatusCode: utils.OK,
+		},
+		{
+			// Error while parsing log
+			name:               "failed parse",
+			msg:                amqp.Delivery{},
+			mockFunc:           func(m *usecase.MockIInsertCTRLogUseCase) {},
+			expectedStatusCode: utils.INVALID_ARGUMENT,
+		},
+		{
+			// Error while inserting log
+			name: "failed insert",
+			msg:  msg,
+			mockFunc: func(m *usecase.MockIInsertCTRLogUseCase) {
+				m.EXPECT().InsertCTRLog(
+					gomock.Any(),
+					&usecase.InsertCTRLogDto{
+						EventType: "event",
+						CreatedAt: fixedTime,
+						ObjectID:  "1234",
+					},
+				).Times(1).Return(errors.New("failed to insert CTR log."))
+			},
+			expectedStatusCode: utils.INTERNAL,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctrl := gomock.NewController(t)
+			defer ctrl.Finish()
+
+			mockInsertUseCase := usecase.NewMockIInsertCTRLogUseCase(ctrl)
+			tt.mockFunc(mockInsertUseCase)
+
+			handler := NewAMQPCTRLogHandler(mockInsertUseCase, &amqp.Channel{})
+
+			err := handler.HandleCTRLog(tt.msg)
+
+			gotStatusCode := errorToStatusCode(err)
+			if gotStatusCode != tt.expectedStatusCode {
+				t.Errorf("Expected status code %d, got %d", tt.expectedStatusCode, gotStatusCode)
+			}
+		})
+	}
+}
+
+func errorToStatusCode(err error) int {
```

Comment:
Writing code as if a response is defined, despite not actually defining one, is not good from a readability perspective.  
Please rewrite the test to explicitly define the expected output for a given input.  
Obviously, since the test is meant to verify whether the expected error is returned, using status codes for validation is incorrect.